mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/DeletedMessageWithStorageInformationDTO.java

```diff
@@ -0,0 +1,272 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.dto;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.vault.DeletedMessage;
+import org.apache.james.vault.metadata.DeletedMessageWithStorageInformation;
+import org.apache.james.vault.metadata.StorageInformation;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+public class DeletedMessageWithStorageInformationDTO {
+
+    public static class StorageInformationDTO {
+
+        public static StorageInformationDTO from(StorageInformation storageInformation) {
+            return new StorageInformationDTO(storageInformation.getBucketName().asString(),
+                storageInformation.getBlobId().asString());
+        }
+
+        private final String bucketName;
+        private final String blobId;
+
+        @JsonCreator
+        public StorageInformationDTO(@JsonProperty("bucketName") String bucketName,
+                                     @JsonProperty("blobId") String blobId) {
+            Preconditions.checkNotNull(bucketName);
+            Preconditions.checkNotNull(blobId);
+
+            this.bucketName = bucketName;
+            this.blobId = blobId;
+        }
+
+        public String getBucketName() {
+            return bucketName;
+        }
+
+        public String getBlobId() {
+            return blobId;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
```

Comment:
do you need equals/hashcode for a DTO?

Reply:
Needed it for the test, to compare the DTOs. Is that wrong?

Reply:
> Needed it for the tests, to compare the DTOs. Is that wrong?

We do compare domain object, we do compare JSON generated from DTOs, there's no reason to compare the DTOs directly?